cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -48,8 +48,8 @@
   private final double _monitoredPartitionsPercentage;
   private final double[] _clusterCapacity;
   private Load _load;
-  // An integer to keep track of the maximum replication factor that a partition was ever created with.
-  private int _maxReplicationFactor;
+  // The replication factor that each topic in the cluster created with.
+  private Map<String, Integer> _replicationFactorByTopic;
```

Comment:
We could keep the maxReplicationFactor variable as well to avoid potential iteration.